__tests__/ExpensiMark-test.js

```diff
@@ -283,3 +283,26 @@ test('Test general email link with various styles', () => {
 
     expect(parser.replace(testString)).toBe(resultString);
 });
+
+test('Test link with brackets', () => {
+    const testString = '[google](http://google.com/(something)?after=parens) '
+        + '([google](http://google.com/(something)?after=parens)) '
+        + '([google](https://google.com/)) '
+        + '([google](http://google.com/(something)?after=parens)))) '
+        + '((([google](http://google.com/(something)?after=parens) '
+        + '(http://foo.com/(something)?after=parens) '
+        + '(((http://foo.com/(something)?after=parens '
+        + '(((http://foo.com/(something)?after=parens))) '
+        + 'http://foo.com/(something)?after=parens))) ';
+
+    const resultString = '<a href="http://google.com/(something)?after=parens" target="_blank">google</a> '
+    + '(<a href="http://google.com/(something)?after=parens" target="_blank">google</a>) '
+    + '(<a href="https://google.com/" target="_blank">google</a>) '
+    + '(<a href="http://google.com/(something)?after=parens" target="_blank">google</a>))) '
+    + '(((<a href="http://google.com/(something)?after=parens" target="_blank">google</a> '
+    + '(<a href="http://foo.com/(something)?after=parens" target="_blank">http://foo.com/(something)?after=parens</a>) '
+    + '(((<a href="http://foo.com/(something)?after=parens" target="_blank">http://foo.com/(something)?after=parens</a> '
+    + '(((<a href="http://foo.com/(something)?after=parens" target="_blank">http://foo.com/(something)?after=parens</a>))) '
+    + '<a href="http://foo.com/(something)?after=parens" target="_blank">http://foo.com/(something)?after=parens</a>))) ';
```

Comment:
Sorry to be super picky...

```suggestion
        + '(<a href="http://google.com/(something)?after=parens" target="_blank">google</a>) '
        + '(<a href="https://google.com/" target="_blank">google</a>) '
        + '(<a href="http://google.com/(something)?after=parens" target="_blank">google</a>))) '
        + '(((<a href="http://google.com/(something)?after=parens" target="_blank">google</a> '
        + '(<a href="http://foo.com/(something)?after=parens" target="_blank">http://foo.com/(something)?after=parens</a>) '
        + '(((<a href="http://foo.com/(something)?after=parens" target="_blank">http://foo.com/(something)?after=parens</a> '
        + '(((<a href="http://foo.com/(something)?after=parens" target="_blank">http://foo.com/(something)?after=parens</a>))) '
        + '<a href="http://foo.com/(something)?after=parens" target="_blank">http://foo.com/(something)?after=parens</a>))) ';
    
```